backend/pipeline/pipeline_processor.py

```diff
@@ -26,23 +27,36 @@ def initialize_pipeline_sync(pipeline_id: str) -> Pipeline:
         )
 
     @staticmethod
-    def fetch_pipeline(pipeline_id: str, check_active: bool = True) -> Pipeline:
+    def fetch_pipeline(
+        pipeline_id: str, check_active: bool = True, created_user: Optional[User] = None
+    ) -> Pipeline:
         """Retrieves and checks for an active pipeline.
+        Args:
+            pipeline_id (str): UUID of the pipeline
+            check_active (bool): Whether to check if the pipeline is active
+            created_user (Optional[User]): User who created the pipeline
 
         Raises:
             InactivePipelineError: If an active pipeline is not found
         """
-        pipeline: Pipeline = Pipeline.objects.get(pk=pipeline_id)
+        if created_user:
+            pipeline: Pipeline = Pipeline.objects.get(
+                pk=pipeline_id, created_by=created_user
+            )
+        else:
+            pipeline: Pipeline = Pipeline.objects.get(pk=pipeline_id)
```

Comment:
@muhammad-ali-e what is the point of adding the created_by check if in the else we are allowing other wise also?